src/dynamic_loading_c_impl.rs

```diff
@@ -131,17 +133,19 @@ impl<T> CKBDLContext<T> {
         if ((size >> RISCV_PGSIZE_SHIFT) << RISCV_PGSIZE_SHIFT) != size {
             return Err(Error::InvalidAlign);
         }
-
+        let from_type_id = match hash_type {
```

Comment:
https://github.com/nervosnetwork/ckb/blob/develop/util/jsonrpc-types/src/blockchain.rs#L21

```
let hash_type = ...
```

The new interface should accept `ScriptHashType` instead of specific to `type_id`.

Reply:
it's not hash_type indeed. It's a boolean value indicating `from type id or not`.